src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -371,13 +458,89 @@ asf::auto_release_ptr<asr::Project> AppleseedInteractiveRender::prepare_project(
             renderer_settings,
             m_bitmap,
             time,
-            m_progress_cb));
+            m_progress_cb,
+            m_object_instance_map,
+            m_material_map));
 
     std::setlocale(LC_ALL, previous_locale.c_str());
 
     return project;
 }
 
+void AppleseedInteractiveRender::assign_material(Mtl* mtl, INode* node)
+{
+    if (mtl == nullptr)
+        return;
+
+    std::vector<Mtl*> materials;
+    const int submtlcount = mtl->NumSubMtls();
+    if (mtl->IsMultiMtl() && submtlcount > 0)
+    {
+        for (int i = 0; i < submtlcount; ++i)
+        {
+            Mtl* submtl = mtl->GetSubMtl(i);
+            materials.push_back(submtl);
+        }
+    }
+    else
+    {
+        materials.push_back(mtl);
+    }
+
+    MaterialMap assigned_materials;
+    for (const auto& mtl : materials)
+    {
+        IAppleseedMtl* appleseed_mtl =
+            static_cast<IAppleseedMtl*>(mtl->GetInterface(IAppleseedMtl::interface_id()));
+        if (appleseed_mtl == nullptr)
+            continue;
+        
+        assigned_materials[mtl] = "";
+    }
+    
+    auto instance_map = m_object_instance_map.find(node->GetObjectRef());
+    if (instance_map != m_object_instance_map.end())
+        get_render_session()->schedule_assign_material(assigned_materials, instance_map->second);
+}
+
+void AppleseedInteractiveRender::update_material(Mtl* mtl)
+{
+    if (mtl == nullptr)
```

Comment:
Is it legal to call `update_material()` with a null argument? If not then replace this by an assertion.